Makefile

```diff
@@ -1,11 +1,198 @@
+endif
+endif
+
+ifndef GIT_REPO_DIR
+	@$(eval GIT_REPO_DIR=~/git)
+endif
+
+ifdef SELINUX_ENABLED
+ifeq ($(shell test "$(SELINUX_ENABLED)" = True  -o  \
+                   "$(SELINUX_ENABLED)" = true && printf "true"), true)
+		@$(eval export SELINUX_ENABLED=,Z)
+endif
+endif
+	@echo
+
+
+## Check external and internal links
+check_links: | envvar stop
+	@echo "${GREEN}Makefile: Check external and internal links${RESET}"
+	${DEBUG}export IFS=$$'\n'; \
+	${CONTAINER_ENGINE} run -it --rm --name userguide -v ${PWD}:/srv:ro${SELINUX_ENABLED} --mount type=tmpfs,destination=/srv/site userguide /bin/bash -c 'cd /srv; bundle install --quiet; rake -- -u'
+	@echo
+
+
+## Check spelling on site content
+check_spelling: | envvar stop
+	@echo "${GREEN}Makefile: Check spelling on site content${RESET}"
+	@echo "Dictionary file: ${GIT_REPO_DIR}/project-infra/images/yaspeller/.yaspeller.json"
+	${DEBUG}export IFS=$$'\n'; \
+	for i in `${CONTAINER_ENGINE} run -it --rm --name yaspeller -v ./docs:/srv/docs:ro${SELINUX_ENABLED} -v ${GIT_REPO_DIR}/project-infra/images/yaspeller/.yaspeller.json:/srv/.yaspeller.json:ro${SELINUX_ENABLED} -v /dev/null:/srv/Gemfile.lock yaspeller /bin/bash -c 'echo; yaspeller -c /srv/.yaspeller.json --only-errors --ignore-tags iframe,img,code,kbd,object,samp,script,style,var /srv/docs'`; do \
+		if [[ "$${i}" =~ "✗" ]]; then \
+			export RETVAL=1; \
+		fi; \
+		echo "$${i}"; \
+	done; \
+	if [ "$${RETVAL}" ]; then exit 1; fi && echo && echo "Complete!"
+
+
+## Build image: casperjs
+build_image_casperjs: stop_casperjs
+	${DEBUG}$(eval export DIR=${GIT_REPO_DIR}/project-infra/images/casperjs)
+	${DEBUG}$(eval export TAG=localhost/casperjs:latest)
+	${DEBUG}$(MAKE) build_image
+	@echo
+
+
+## Build image: userguide
+build_image_userguide: stop
+	${DEBUG}$(eval export DIR=${GIT_REPO_DIR}/project-infra/images/kubevirt-userguide)
+	${DEBUG}$(eval export TAG=localhost/userguide:latest)
+	${DEBUG}$(MAKE) build_image
+	@echo
+
+
+## Build image: yaspeller
+build_image_yaspeller: stop_yaspeller
+	${DEBUG}$(eval export DIR=${GIT_REPO_DIR}/project-infra/images/yaspeller)
+	${DEBUG}$(eval export TAG=localhost/yaspeller:latest)
+	${DEBUG}$(MAKE) build_image
+	@echo
+
+
+build_image: envvar
+	@echo "${GREEN}Makefile: Building image: ${TAG}${RESET}"
+ifeq ($(DIR),)
+	@echo "This is a sourced target!"
+	@echo "Do not run this target directly... exitting!"
+	exit 1
+endif
+	cd ${DIR} && \
+	(${CONTAINER_ENGINE} rmi ${TAG} || echo -n) && \
+	${BUILD_ENGINE} ${TAG}
+
+
+## Build site. This target should only be used by Prow jobs.
+build: envvar
+	@echo "${GREEN}Makefile: Build mkdocs site.  This should only be used by a Prow job.${RESET}"
+	if [ `which python3.7` ]; then \
+		python3.7 -m venv /tmp/venv; \
+		. /tmp/venv/bin/activate; \
+		pip3 install mkdocs mkdocs-awesome-pages-plugin mkdocs-htmlproofer-plugin; \
+		echo '*** BEGIN mkdocs.yml ***'; \
+		cat mkdocs.yml; \
+		echo '*** END mkdocs.yml ***'; \
+		mkdocs build -f mkdocs.yml -d site; \
```

Comment:
What happens after this? how does the rendered website make it into the branch that gets served?

Reply:
@dhiller  Correct me if I'm wrong please.  The tide job handles the merge.

Reply:
Yes, the tide job will handle merging the PR into the target branch. But IIUC the docs of mkdocs state that [there's a separate step required](https://metager.de/meta/meta.ger3?eingabe=mkdocs+gh-deploy) to actually deploy your site to GH Pages. 

I always understood netlify only being responsible for showing the previews of PRs to merge, if it also handles display of the merged files then we should be good. If it doesn't deploy the rendered site to gh pages, we need to add another make target that does execute the deployment and adjust the periodic job so that it calls the deploy make target instead.

Makes sense?